### PR TITLE
[feat/aut834] Include/Pass version tag to SmartPeak code.

### PR DESCRIPTION
**Pull request by ChristopherAbram**
+ Modify CMakeLists.txt to cache version information.
+ Add `Utilities::getSmartPeakVersion()` static method.
+ Include version information in GUI.cpp.

📌 I think this method doesn't need test.

### CODE REVIEW

File: src/examples/source/GUI.cpp [hunk 1]
```diff
@@ -381,8 +381,9 @@ int main(int argc, char** argv)
     }
     if (ImGui::BeginPopupModal("About modal", NULL, ImGuiWindowFlags_AlwaysAutoResize))
     {
+      // TODO: read text from file and print in the window
```
Comment (bertrandboudaud) [inline src/examples/source/GUI.cpp:393]: Would be better to add a task in JIRA than getting a TODO inside the code, for this kind of remark. TODO goes well for missing states handling, or identified performance issues while writing the code, for example. everything that would help someone debugging there. features should go to to JIRA. This is minor :)

**Review by bertrandboudaud — APPROVED** [review]
Looks good to me.
another place where the version should be displayed, is on the log, on startup, to ease debugging investigation based on the log.

File: CMakeLists.txt [hunk 2]
```diff
@@ -78,9 +78,9 @@ list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/Windows")
 #------------------------------------------------------------------------------
 # SmartPeak Version Information
 #------------------------------------------------------------------------------
-set (SMARTPEAK_PACKAGE_VERSION_MAJOR 1)
-set (SMARTPEAK_PACKAGE_VERSION_MINOR 0)
-set (SMARTPEAK_PACKAGE_VERSION_PATCH 0)
+set(SMARTPEAK_PACKAGE_VERSION_MAJOR "0" CACHE STRING "SmartPeak major version")
+set(SMARTPEAK_PACKAGE_VERSION_MINOR "0" CACHE STRING "SmartPeak minor version")
+set(SMARTPEAK_PACKAGE_VERSION_PATCH "0" CACHE STRING "SmartPeak patch version")
```
Comment (ghost) [inline CMakeLists.txt]: how are the version numbers carried out to the respective variables?

File: src/examples/source/GUI.cpp (hunk 1 above)
Comment (ChristopherAbram) [reply src/examples/source/GUI.cpp:393]: This TODO is more of a note for a future when we decide to make GUI modular - at the moment it can stay as it is

**Comment by ChristopherAbram** [comment]
> Looks good to me.
> another place where the version should be displayed, is on the log, on startup, to ease debugging investigation based on the log.

Good point, I will append it to the log 👍

**Review by ghost — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (ChristopherAbram) [reply CMakeLists.txt]: I think we will pass these variables while lunching cmake config. By default version is `0.0.0` which means no version defined

Comment (ChristopherAbram) [reply CMakeLists.txt]: Please elaborate what you would like to modify

File: src/examples/source/GUI.cpp [hunk 3]
```diff
@@ -158,6 +158,9 @@ int main(int argc, char** argv)
     .addAppender(&consoleAppender)
     .addAppender(&appender_);
 
+  // Log SmartPeak launch initiated:
+  LOG_INFO << "Start SmartPeak version " << Utilities::getSmartPeakVersion();
```
Comment (ChristopherAbram) [inline src/examples/source/GUI.cpp:162]: I hope this satisfies your request @bertrandboudaud

File: CMakeLists.txt (hunk 2 above)
Comment (ghost) [reply CMakeLists.txt]: > I think we will pass these variables while lunching cmake config. By default version is `0.0.0` which means no version defined

could you provide a link to the cmake config file you are referring to?:)

**Comment by ChristopherAbram** [comment]
I have addressed @ahmedskhalil issue and also changed title bar for SmartPeak - now it also includes version.

**Review by ghost — APPROVED** [review]
very nice

**Pull request merged**